tests/ui/formulaBar.test.ts

```diff
@@ -0,0 +1,180 @@
+// Import other necessary dependencies and utilities for testing
+import LightSheet from "../../src/main";
+import UI from "../../src/ui/render";
+describe("Formula", () => {
+  let targetElementMock: HTMLElement;
+  let lightSheetInstance: LightSheet;
+  beforeEach(() => {
+    targetElementMock = document.createElement("div");
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test("Exist formula bar in dom", () => {
+    lightSheetInstance = new LightSheet(targetElementMock, {
+      data: [],
+    });
+    new UI(targetElementMock, lightSheetInstance);
+    // Find the formula bar container
+    const formulaBarContainer = targetElementMock.querySelector(
+      ".lightsheet_table_formula_bar",
+    );
+
+    // Assert that the formula bar container exists in the DOM
+    expect(formulaBarContainer).toBeTruthy();
+
+    // Assert that the container has an input field with the class "formula_input"
+    const formulaInputField =
+      formulaBarContainer?.querySelector(".formula_input"); // Using optional chaining
+    expect(formulaInputField).toBeTruthy(); // Using a truthy assertion instead
+  });
+
+  test("Change cell content from formula bar", () => {
+    lightSheetInstance = new LightSheet(targetElementMock, {
```

Comment:
if you are creating a new instance of LightSheet everytime. instead of reusing the lightsheet variable, you want to have a new const lightsheet variable. example: Const lightSheetInstance = new LightSheet(....)

Reply:
if you dont need to keep to lightsheet object you can just remove it